lib/puma/cluster.rb

```diff
@@ -474,6 +491,29 @@ def run
 
       @launcher.events.fire_on_booted!
 
+      Thread.new do
+        Puma.set_thread_name "acceptor"
+
+        # Chooses a load-balancing algorithm from a set of predefined symbol constants or a custom Proc.
+        algorithms = {
+          least_busy: lambda {|workers| workers.min_by {|w| [-w.capacity, w.last_request]}},
+          round_robin: lambda {|workers| workers.min_by(&:last_request)},
+          pile_up: lambda {|workers| workers.select {|w| w.capacity > 0}.min_by {|w| w.index} || workers.min_by(&:last_request)},
+          random: lambda {|workers| workers.sample}
+        }.freeze
```

Comment:
This should be moved out of here into a constant rather than hidden here, probably even in it's own file.